Mod compatibility: Borderless Mining 1.1.7 (1.19.4) and 1.1.6 (1.19.3) cause a crash when attempting to open the Options menu while OptiFabric 1.13.24 is also installed.

**Mod compatibility:** Borderless Mining 1.1.7 (1.19.4) and 1.1.6 (1.19.3) cause a crash when attempting to open the Options menu while OptiFabric 1.13.24 is also installed.

This occurs with a minimal mod set consisting of only:
* Fabric API
	* `0.76.0+1.19.4` (1.19.4)
	* `0.76.0+1.19.3` (1.19.3)
* Borderless Mining
	* `1.1.7+1.19.4` (1.19.4)
	* `1.1.6+1.19.3` (1.19.3)
* OptiFabric `1.13.24`
* OptiFine
	* 1.19.4 HD U I4 pre1
	* 1.19.4 HD U I3
	* 1.19.3 HD U I3

## Stack trace (1.19.4)
```
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:545) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:169) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:118) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:169) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:?]
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3474) ~[lwjgl-glfw-3.3.1.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:245) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1241) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:801) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:237) ~[fabric-loader-0.14.18-1.19.4.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.18.jar:?]
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_446 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.18.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_442.method_19863(class_442.java:179) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:57) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:187) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[client-intermediary.jar:?]
	at net.minecraft.class_442.method_25402(class_442.java:431) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:540) ~[client-intermediary.jar:?]
	... 14 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.18.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_442.method_19863(class_442.java:179) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:57) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:187) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[client-intermediary.jar:?]
	at net.minecraft.class_442.method_25402(class_442.java:431) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:540) ~[client-intermediary.jar:?]
	... 14 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Multi-argument modifier method modifyOption(Lorg/spongepowered/asm/mixin/injection/invoke/arg/Args;)V in borderlessmining.mixins.json:FullScreenOptionMixin from mod borderlessmining failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap borderless-mining-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1062) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:402) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:327) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.18.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_442.method_19863(class_442.java:179) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:57) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:187) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[client-intermediary.jar:?]
	at net.minecraft.class_442.method_25402(class_442.java:431) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:540) ~[client-intermediary.jar:?]
	... 14 more
```

## Stack trace (1.19.3)
```
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:497) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:165) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:118) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:165) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:?]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3403) ~[lwjgl-glfw-3.3.1.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:199) ~[client-intermediary.jar:?]
	at net.minecraft.class_1041.method_15998(class_1041.java:396) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1222) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:781) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:244) ~[fabric-loader-0.14.18-1.19.3.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[fabric-loader-0.14.18-1.19.3.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.18.jar:?]
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_446 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.18.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_442.method_19863(class_442.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:16) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:172) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:31) ~[client-intermediary.jar:?]
	at net.minecraft.class_442.method_25402(class_442.java:457) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:492) ~[client-intermediary.jar:?]
	... 16 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.18.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_442.method_19863(class_442.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:16) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:172) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:31) ~[client-intermediary.jar:?]
	at net.minecraft.class_442.method_25402(class_442.java:457) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:492) ~[client-intermediary.jar:?]
	... 16 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Multi-argument modifier method modifyOption(Lorg/spongepowered/asm/mixin/injection/invoke/arg/Args;)V in borderlessmining.mixins.json:FullScreenOptionMixin from mod borderlessmining failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap borderless-mining-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1062) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:402) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:327) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.18.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.18.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_442.method_19863(class_442.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:16) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:172) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:31) ~[client-intermediary.jar:?]
	at net.minecraft.class_442.method_25402(class_442.java:457) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:492) ~[client-intermediary.jar:?]
	... 16 more
```

--- a/src/main/java/me/modmuss50/optifabric/compat/borderlessmining/BorderlessMiningMixinPlugin.java
+++ b/src/main/java/me/modmuss50/optifabric/compat/borderlessmining/BorderlessMiningMixinPlugin.java
@@ -23,7 +23,7 @@ public class BorderlessMiningMixinPlugin extends InterceptingMixinPlugin {
 
 			for (MethodNode method : targetClass.methods) {
 				if (init.equals(method.name) && "()V".equals(method.desc)) {
-					Member createSimpleOption = RemappingUtils.mapMethod("class_7172", "<init>", "(Ljava/lang/String;Lnet/minecraft/class_7172$class_7307;Lnet/minecraft/class_7172$class_7303;Lnet/minecraft/class_7172$class_7178;Ljava/lang/Object;Ljava/util/function/Consumer;)V");
+					Member createSimpleOption = RemappingUtils.mapMethod("class_7172", "<init>", getCreateSimpleOptionInitDescriptor());
 					InsnList extra = new InsnList();
 					LabelNode skip = new LabelNode();
 
@@ -38,5 +38,9 @@ public class BorderlessMiningMixinPlugin extends InterceptingMixinPlugin {
 		}
 
 		super.preApply(targetClassName, targetClass, mixinClassName, mixinInfo);
+	}
+
+	protected String getCreateSimpleOptionInitDescriptor() {
+		return "(Ljava/lang/String;Lnet/minecraft/class_7172$class_7307;Lnet/minecraft/class_7172$class_7303;Lnet/minecraft/class_7172$class_7178;Ljava/lang/Object;Ljava/util/function/Consumer;)V";
 	}
 }

--- a/src/main/java/me/modmuss50/optifabric/compat/borderlessmining/BorderlessMiningNewMixinPlugin.java
+++ b/src/main/java/me/modmuss50/optifabric/compat/borderlessmining/BorderlessMiningNewMixinPlugin.java
@@ -1,0 +1,8 @@
+package me.modmuss50.optifabric.compat.borderlessmining;
+
+public class BorderlessMiningNewMixinPlugin extends BorderlessMiningMixinPlugin {
+    @Override
+    protected String getCreateSimpleOptionInitDescriptor() {
+        return "(Ljava/lang/String;Lnet/minecraft/class_7172$class_7277;Lnet/minecraft/class_7172$class_7303;Lnet/minecraft/class_7172$class_7178;Ljava/lang/Object;Ljava/util/function/Consumer;)V";
+    }
+}

--- a/src/main/java/me/modmuss50/optifabric/compat/borderlessmining/mixin/OptionFullscreenResolutionNewMixin.java
+++ b/src/main/java/me/modmuss50/optifabric/compat/borderlessmining/mixin/OptionFullscreenResolutionNewMixin.java
@@ -1,0 +1,22 @@
+package me.modmuss50.optifabric.compat.borderlessmining.mixin;
+
+import me.modmuss50.optifabric.compat.borderlessmining.MagicMixinBridge;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.option.VideoOptionsScreen;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Pseudo;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyArgs;
+import org.spongepowered.asm.mixin.injection.invoke.arg.Args;
+
+@Pseudo
+@Mixin(targets = "net.optifine.gui.OptionFullscreenResolution", remap = false)
+abstract class OptionFullscreenResolutionNewMixin {
+	@ModifyArgs(method = "make",
+				at = @At(value = "INVOKE", target = "Lnet/minecraft/class_7172;<init>(Ljava/lang/String;Lnet/minecraft/class_7172$class_7307;Lnet/minecraft/class_7172$class_7277;Lnet/minecraft/class_7172$class_7178;Ljava/lang/Object;Ljava/util/function/Consumer;)V", remap = true))
+	private static void modifyOption(Args args) {
+		Screen screen = MinecraftClient.getInstance().currentScreen;
+		((MagicMixinBridge) (screen instanceof MagicMixinBridge ? screen : new VideoOptionsScreen(screen, MinecraftClient.getInstance().options))).optifabric£modifyFullscreenButton(args);
+	}
+}

--- a/src/main/java/me/modmuss50/optifabric/mod/OptifabricSetup.java
+++ b/src/main/java/me/modmuss50/optifabric/mod/OptifabricSetup.java
@@ -177,7 +177,7 @@ public class OptifabricSetup implements Runnable {
 
 					for (MethodNode method : node.methods) {
 						if ("renderBatched".equals(method.name) && method.desc.endsWith(desc)) {
-							Mixins.addConfiguration("optifabric.compat.indigo.newer-mixins.json");							
+							Mixins.addConfiguration("optifabric.compat.indigo.newer-mixins.json");
 							return;
 						}
 					}
@@ -235,7 +235,7 @@ public class OptifabricSetup implements Runnable {
 			Mixins.addConfiguration("optifabric.optifine.old-mixins.json");
 		}
 
-        if (isPresent("fabricloader", ">=0.13.0") && (isPresent("cloth-client-events-v0", ">=3.1.58") || isPresent("cloth-client-events-v0", ">=2.1.60 <3.0") || isPresent("cloth-client-events-v0", ">=1.6.59 <2.0"))) {
+		if (isPresent("fabricloader", ">=0.13.0") && (isPresent("cloth-client-events-v0", ">=3.1.58") || isPresent("cloth-client-events-v0", ">=2.1.60 <3.0") || isPresent("cloth-client-events-v0", ">=1.6.59 <2.0"))) {
 			// no mixins are needed -- cloth had a workaround for https://github.com/FabricMC/Mixin/issues/80
 			// but it is now fixed in fabricloader
 		} else if (isPresent("cloth-client-events-v0", ">=2.0")) {
@@ -377,7 +377,7 @@ public class OptifabricSetup implements Runnable {
 						}
 					}
 				});
-			}	
+			}
 		} else if (isPresent("charm", ">=3.0 <4.0")) {
 			Mixins.addConfiguration("optifabric.compat.charm.mixins.json");
 		} else if (isPresent("charm", ">=4.0")) {
@@ -510,7 +510,9 @@ public class OptifabricSetup implements Runnable {
 			Mixins.addConfiguration("optifabric.compat.zoomify.mixins.json");
 		}
 
-		if (isPresent("borderlessmining", ">=1.1.3")) {
+		if (isPresent("borderlessmining", ">=1.1.6")) {
+			Mixins.addConfiguration("optifabric.compat.borderlessmining.new-mixins.json");
+		} else if (isPresent("borderlessmining", ">=1.1.3")) {
 			Mixins.addConfiguration("optifabric.compat.borderlessmining.mixins.json");
 		}
 

--- a/src/main/resources/optifabric.compat.borderlessmining.new-mixins.json
+++ b/src/main/resources/optifabric.compat.borderlessmining.new-mixins.json
@@ -1,0 +1,9 @@
+{
+	"parent": "optifabric.mixins.json",
+	"package": "me.modmuss50.optifabric.compat.borderlessmining.mixin",
+	"plugin": "me.modmuss50.optifabric.compat.borderlessmining.BorderlessMiningNewMixinPlugin",
+	"mixins": [
+		"VideoOptionsScreenMixin",
+		"OptionFullscreenResolutionNewMixin"
+	]
+}

